### DISCUSSION TITLE
How do I install a framework that is not currently supported by bun create?

### DISCUSSION THREAD

**Discussion by chebread** [oven-sh/bun#1995] [Q&A]
### What version of Bun is running?

0.5.0

### What platform is your computer?

Darwin 22.2.0 x86_64 i386

## Bug
I tried `bunx create-remix test`.
I chose options from `Just the basics`, `Cloudflare Pages`, `JavaScript`, `Y` (do you use bun install?).

Then `error: script not found "config"
Command failed: bun config get @remix-run:registry
An error: script not found "config"` occurred,

After that, I entered the `bun install` command to download the node packages.

Also, I ran the development server with `bun run dev`.
Then I got esbuild related errors.

```
❯ bun run dev
$ remix build && run-p \"dev:*\"
Building Remix app in production mode...

You installed esbuild for another platform than the one you're currently using.
This won't work because esbuild is written with native code and needs to
install a platform-specific binary executable.

Specifically the "@esbuild/darwin-arm64" package is present but this platform
needs the "@esbuild/darwin-x64" package instead. People often get into this
situation by installing esbuild with npm running inside of Rosetta 2 and then
trying to use it with node running outside of Rosetta 2, or vice versa (Rosetta
2 is Apple's on-the-fly x86_64-to-arm64 translation service).

If you are installing with npm, you can try ensuring that both npm and node are
not running under Rosetta 2 and then reinstalling esbuild. This likely involves
changing how you installed npm and/or node. For example, installing node with
the universal installer here should work: https://nodejs.org/en/download/. Or
you could consider using yarn instead of npm which has built-in support for
installing a package on multiple platforms simultaneously.

If you are installing with yarn, you can try listing both "arm64" and "x64"
in your ".yarnrc.yml" file using the "supportedArchitectures" feature:
https://yarnpkg.com/configuration/yarnrc/#supportedArchitectures
Keep in mind that this means multiple copies of esbuild will be present.

Another alternative is to use the "esbuild-wasm" package instead, which works
the same way on all platforms. But it comes with a heavy performance cost and
can sometimes be 10x slower than the "esbuild" package, so you may also not
want to do that.

Error
    at Object.onCompileFailure (test/node_modules/@remix-run/dev/dist/cli/commands.js:179:13)
    at Object.compile (test/node_modules/@remix-run/dev/dist/compiler/remixCompiler.js:33:134)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Object.build (test/node_modules/@remix-run/dev/dist/compiler/build.js:33:3)
    at async Object.build (test/node_modules/@remix-run/dev/dist/cli/commands.js:174:3)
    at async Object.run (test/node_modules/@remix-run/dev/dist/cli/run.js:448:7)
error: script "dev" exited with code 1 (SIGHUP)
```

How can I configure a remix app using bun and bunx?

**Comment by Jarred-Sumner** [comment]
We fixed some bugs related to `bunx` very recently

It looks like this ends up failing due to running a comamnd we don't support:
<img width="779" alt="image" src="https://user-images.githubusercontent.com/709451/217685615-cadc2aa5-60f2-462f-a219-edafec2a2c5a.png">

@jacob-ebey it looks like it's running `bun config get @remix-run:registry`? maybe that could be skipped for bun?

**Reply by hjonasson** [reply]
I am looking at running Remix with Bun. The docs currently say

> Remix currently relies on Node.js APIs that Bun does not yet implement

[From here](https://github.com/oven-sh/bun/blob/main/docs/guides/ecosystem/remix.md)

Were you referring to this discussion thread @colinhacks when you wrote that page? If not, would you mind pointing me to what is missing? In general, Remix isn't (shouldn't) be tied to Node so I think this should be doable but I would love to know what has blocked the full use of Bun with Remix in the past.
